pkg/api/v20191231preview/openshiftcluster_validatestatic.go

```diff
@@ -86,20 +86,24 @@ func (sv *openShiftClusterStaticValidator) validateProperties(path string, p *Op
 	if err := sv.validateConsoleProfile(path+".consoleProfile", &p.ConsoleProfile); err != nil {
 		return err
 	}
-	if err := sv.validateServicePrincipalProfile(path+".servicePrincipalProfile", &p.ServicePrincipalProfile); err != nil {
-		return err
+	if isCreate {
```

Comment:
going to rebase and revert this change, now that servicePrincipalProfile is no longer dynamic.